File: streaming-service/src/main/java/io/spring/demo/streaming/portfolio/UserController.java
package io.spring.demo.streaming.portfolio;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

	private final UserRepository userRepository;

	public UserController(UserRepository userRepository) {
		this.userRepository = userRepository;
	}

	@GetMapping("/users")
	public Flux<User> fetchAll() {
		return this.userRepository.findAll();
	}

	@GetMapping("/users/{github}")
	public Mono<User> fetchUser(@PathVariable String github) {
		return this.userRepository.findUserByGithub(github);
	}
}

File: streaming-service/src/main/kotlin/io/spring/demo/streaming/portfolio/UserController.kt
package io.spring.demo.streaming.portfolio;

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(val userRepository: UserRepository) {

	@GetMapping("/users")
	fun fetchAll() = userRepository.findAll()

	@GetMapping("/users/{github}")
	fun fetchUser(@PathVariable github: String) = userRepository.findUserByGithub(github)

}
